wallet-core/wallet-operations.md

```diff
@@ -0,0 +1,182 @@
+
+```swift
+let wallet = HDWallet(strength: 128, passphrase: "")
+```
+
+Input parameter | Description
+---|---
+*strength* | The strength of the secret seed.  Higher seed means more information content, longer recovery phrase.  Default value is **128**, but 256 is also possible.
+*passphrase* | Optional passphrase, used to scramble the seed.  If specified, the wallet can be imported and opened only with the passphrase (Not to be confused with recovery phrase).
+
+### Importing a Multi-Coin Wallet
+
+A previously created wallet can be imported using the recovery phrase.  Typical usecases for import are:
+* re-importing a wallet later, into a later installation, or
+* importing into another device, or 
+* importing into another wallet app.
+
+If the wallet was created with a passphrase, it is also required.
+
+```swift
+let wallet = HDWallet(mnemonic: "ripple scissors kick mammal hire column oak again sun offer wealth tomorrow wagon turn fatal", passphrase: "")
+```
+
+Input parameter | Description
+---|---
+*mnemonic* | a.k.a. *recovery phrase*.  The string of several words that was used to create the wallet.
+*passphrase* | Optional passphrase, used to encrypt the seed.
+
+
+## Account Address Derivation
+
+Each coin needs a different account, with matching address.  Addresses are derived from the multi-coin wallet.
+Derivation is based on a *derivation path*, which is unique for each coin, but can have other parameters as well.
+Each coin has a default derivation path, such as `"m/84'/0'/0'/0/0"` for Bitcoin and `"m/44'/60'/0'/0/0"` for Ethereum.
+
+### Generating the Default Address for a Coin
+
+The simplest is to get the default address for a coin -- this requires no further inputs.
+The address is generated using the default derivation path of the coin.
+
+For example, the default BTC address, derived for the wallet with the mnemonic shown above, with the default BTC derivation path (`m/84'/0'/0'/0/0`) is:
+`bc1qpsp72plnsqe6e2dvtsetxtww2cz36ztmfxghpd`.
+For Ethereum, this is `0xA3Dcd899C0f3832DFDFed9479a9d828c6A4EB2A7`.
+
+Here is the sample code fort obtaining the default address for different coins:
+
+```swift
+let addressBTC = wallet.getAddressForCoin(coin: .bitcoin)
+let addressETH = wallet.getAddressForCoin(coin: .ethereum)
+let addressBNB = wallet.getAddressForCoin(coin: .binance)
+```
+
+### Generating an Address Using a Custom Derivation Path (Expert)
+
+It is also possible to derive addresses using custom derivation paths.
+This can be done in two steps: first a derived private key is obtained, then an address from it.
+
+> **Warning**: use this only if you are well aware of the semantics of the derivation path used!
+
+> **Security Warning**: if secrets such as private keys are handled by the wallet, even if for a short time, handle with care!
+> Avoid any risk of leakage of secrets!
+
+```swift
+let key = wallet.getKey(derivationPath: "m/44\'/60\'/1\'/0/0")   // m/44'/60'/1'/0/0
```

Comment:
```suggestion
let key = wallet.getKey(derivationPath: "m/44'/60'/1'/0/0")   // m/44'/60'/1'/0/0
```

Reply:
Oddly this did not work for me, it gave an exception that components must be separated by '/'.  I tried escaping the 's, and then it worked.